### PR TITLE
[nrf toup] arch: Allow removing drivers while keeping arch and kernel

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Add dummy interrupt handlers and driver functions.
Rename CONFIG_KERNEL to CONFIG_DRIVERS and have it affect drivers only.

This helps with porting B0 to use more native Zephyr code.

### CODE REVIEW

**Comment by oyvindronningstad** [comment]
cc @sigvartmh @hakonfam @osaether

**Comment by sigvartmh** [comment]
I get the idea, but do we want to have this dummy stuff in the zephyr repo?

File: drivers/Kconfig [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #
 
 
+if KERNEL
```
Comment (hakonfam) [inline drivers/Kconfig]: Isn't this symbol now replaced with `DRIVERS` ?

File: arch/arm/core/CMakeLists.txt [hunk 2]
```diff
@@ -23,6 +23,10 @@ zephyr_library_sources(
   thread_abort.c
 )
 
+if (NOT DEFINED CONFIG_DRIVERS)
```
Comment (hakonfam) [inline arch/arm/core/CMakeLists.txt]: Would `if (NOT CONFIG_DRIVERS)` work?

File: Kconfig.zephyr [hunk 3]
```diff
@@ -50,8 +50,8 @@ menu "Build and Link Features"
 
 menu "Linker Options"
 
-config KERNEL
-	bool "Include Zephyr kernel"
+config DRIVERS
```
Comment (hakonfam) [inline Kconfig.zephyr]: Actually, we don't need to have any kconfig like this. Including the `driver` folder works without compilation error now!

File: CMakeLists.txt [hunk 4]
```diff
@@ -457,7 +457,7 @@ endif()
 add_subdirectory(boards)
 add_subdirectory(ext)
 add_subdirectory(subsys)
-add_subdirectory_ifdef(CONFIG_KERNEL drivers)
+add_subdirectory_ifdef(CONFIG_DRIVERS drivers)
```
Comment (hakonfam) [inline CMakeLists.txt]: Try to make this non-conditional, AFAICS it will work without the kconfig check.

**Comment by oyvindronningstad** [comment]
Not needed now.

**Comment by hakonfam** [comment]
Is closing this related to my PR upstream? I am not sure that this will be accepted.

**Pull request closed without merging**